### ISSUE TITLE
Persist users who login using api_client param

### ISSUE THREAD

**Issue by sbose78** [fabric8-services/fabric8-auth#673]
1. On login, check if `api_client` param is present ( as done now ). If the user doesn't exist in the DB, create a record before returning a token.

[ not critical now ] 
2. Modify `Create User` API to allow updating the user record with the cluster api url.

**Comment by sbose78** [comment]
We could do (1) now and move on. If needed, we could add support for (2) later on.

**Comment by sbryzak** [comment]
@sbose78 Certainly, this would solve the problem of unrevokable tokens, as we would be able to create a token record if the user has a corresponding identity record.

**Comment by xcoulon** [comment]
@sbose78 if step 2 is not critical for now and can be addressed later, shouldn't we open a separate issue for it?

**Comment by sbose78** [comment]
> if step 2 is not critical for now and can be addressed later, shouldn't we open a separate issue for it?
Yeah, if it's not solved in a single PR, we could do that.
